moveit_ros/perception/mesh_filter/src/transform_provider.cpp

```diff
@@ -126,25 +128,26 @@ void TransformProvider::setUpdateInterval(unsigned long usecs)
 
 void TransformProvider::updateTransforms()
 {
-  static Affine3d transformation;
+  static tf2::Stamped<Affine3d> input_transform, output_transform;
   static robot_state::RobotStatePtr robot_state;
   robot_state = psm_->getStateMonitor()->getCurrentState();
 
   static string error;
-  static Stamped<Pose> input_pose, out_pose;
-  tf_->getLatestCommonTime(frame_id_, psm_->getPlanningScene()->getPlanningFrame(), input_pose.stamp_, &error);
-  input_pose.frame_id_ = psm_->getPlanningScene()->getPlanningFrame();
+  tf_buffer_->_getLatestCommonTime( tf_buffer_->_lookupFrameNumber(frame_id_),
+                tf_buffer_->_lookupFrameNumber(psm_->getPlanningScene()->getPlanningFrame()),
+                input_transform.stamp_, &error);
+  input_transform.frame_id_ = psm_->getPlanningScene()->getPlanningFrame();
 
   for (map<MeshHandle, shared_ptr<TransformContext> >::const_iterator contextIt = handle2context_.begin();
        contextIt != handle2context_.end(); ++contextIt)
   {
-    transformation = robot_state->getLinkState(contextIt->second->frame_id_)->getGlobalCollisionBodyTransform();
-    poseEigenToTF(transformation, input_pose);
     try
     {
-      tf_->transformPose(frame_id_, input_pose, out_pose);
+      // TODO: check logic here - which global collision body's transform should be used?
+      input_transform.setData(robot_state->getAttachedBody(contextIt->second->frame_id_)->getGlobalCollisionBodyTransforms()[0]);
```

Comment:
Well, `transfom_provider` is never actually compiled, and if it were to be compiled, it would [introduce a circular dependency](https://github.com/ros-planning/moveit/pull/830/files#diff-be7de8b6a8969d6b0e3c7d204c20b2edL22) between `moveit_ros_perception` and `moveit_ros_planning`, due to its use of `planning_scene_monitor`. 

Additionally, it turns out the function[`getGlobalCollisionBodyTransform` doesn't exist anywhere in the codebase](https://github.com/ros-planning/moveit/search?utf8=%E2%9C%93&q=getGlobalCollisionBodyTransform&type=). In order to make sure all of my other updates to this file were correct (tf2 message & type conversions, etc), I forced the compile of this file, which fails on this line. I put a band-aid on it to allow compiling, but it's likely not correct. I'm not sure what the proper procedure should be with this dead code.